weaviate/logger.py

```diff
@@ -1,5 +1,250 @@
+
+def _setup_logger():
+    """Set up the weaviate-client logger with proper configuration."""
+    logger = logging.getLogger("weaviate-client")
+    log_level = os.getenv("WEAVIATE_LOG_LEVEL", "INFO")
+    logger.setLevel(getattr(logging, log_level))
+    logger.propagate = True  # Ensure log propagation is enabled
+    
+    # Add a StreamHandler if none exists
+    if not logger.handlers:
+        handler = logging.StreamHandler()
+        handler.setLevel(getattr(logging, log_level))
+        formatter = logging.Formatter('%(message)s')
+        handler.setFormatter(formatter)
+        logger.addHandler(handler)
+    
+    print(f"DEBUG: Logger initialized with level {logger.level}, handlers: {logger.handlers}")
+    return logger
+
+logger = _setup_logger()
+
+def _mask_sensitive_headers(headers: Dict[str, str], is_response: bool = False) -> Dict[str, str]:
+    """Mask sensitive information in headers like authorization tokens and cookies.
+    
+    Args:
+        headers: Dictionary of headers to mask
+        is_response: Whether these are response headers (affects which headers to mask)
+    
+    Returns:
+        Dictionary with sensitive headers masked
+    """
+    masked = {}
+    for key, value in headers.items():
+        key_lower = key.lower()
+        if key_lower == "authorization" or "authorization" in key_lower:
+            value = "[...]"
+        elif key_lower == "cookie" and not is_response:
+            value = value.split("=")[0] + "=..."
+        elif key_lower == "set-cookie" and is_response:
+            value = value.split("=")[0] + "=..."
+        elif any(api_key in key_lower for api_key in ["api-key", "api_key", "token", "secret"]):
+            value = "[...]"
+        elif "access-control-allow-headers" in key_lower and "authorization" in value.lower():
+            # Special handling for CORS headers that list Authorization
+            value = "[...]"
+        masked[key] = value
+    return masked
+
+def log_http_event(response: httpx.Response) -> None:
+    """Log HTTP request and response details using the default logger.
+    
+    Args:
+        response: The httpx Response object containing both request and response info
+    """
+    print(f"\nDEBUG: log_http_event called")
+    print(f"DEBUG: Logger state: level={logger.level}, handlers={logger.handlers}, propagate={logger.propagate}")
+    print(f"DEBUG: Effective level: {logger.getEffectiveLevel()}")
+
+    if logger.getEffectiveLevel() > logging.DEBUG:
+        print("DEBUG: Skipping logging due to level")
+        return
+
+    try:
+        request = response.request
+        print(f"DEBUG: Processing request {request.method} {request.url}")
+        
+        # Log request details with clear separation
+        separator = "==================== HTTP Request ===================="
+        print(f"DEBUG: About to log separator: {separator}")
```

Comment:
This is a `print` statement and not `logger.debug` and so will show up regardless of the `WEAVIATE_LOG_LEVEL` specification making the client noisy